api/src/main/java/org/openmrs/module/bedmanagement/dao/BedManagementDao.java

```diff
@@ -38,10 +38,24 @@ public interface BedManagementDao {
 	
 	BedPatientAssignment getBedPatientAssignmentByUuid(String uuid);
 	
+	List<BedPatientAssignment> getBedPatientAssignmentByEncounter(String encunterUuid, boolean includeEnded);
+	
+	List<BedPatientAssignment> getBedPatientAssignmentByVisit(String visitUuid, boolean includeEnded);
+	
 	List<BedPatientAssignment> getCurrentAssignmentsByBed(Bed bed);
 	
 	Bed getLatestBedByVisit(String visitUuid);
 	
+	/**
+	 * Return a list of beds assigned to a patient associated with the given visit.
+	 * 
+	 * @param visitUuid
+	 * @param includeEnded If false, only the currently active bed assignment is returned. Otherwise,
+	 *            all assignments within the visit are returned.
+	 * @return
+	 */
+	List<Bed> getAssignedBedsByVisit(String visitUuid, boolean includeEnded);
+	
```

Comment:
What is the use case for this method vs. the above getBedPatientAssignmentByVisit method?  It seems odd to return a number of Bed objects for a patient vs. BedPatientAssignments...

Reply:
You're right, this can be removed. `unAssignBedsInEndedVisit` is using it but it should be replaceable by `getBedPatientAssignmentByVisit`.